src/go/rpk/pkg/config/params.go

```diff
@@ -552,3 +568,133 @@ func (c *Config) addUnsetDefaults() {
 		r.AdminAPI.Addresses = []string{"127.0.0.1:9644"}
 	}
 }
+
+// Set allow to set a single configuration property by passing a key value pair
+//
+//   Key:    string containing the yaml property tag, e.g: 'rpk.admin_api'.
+//   Value:  string representation of the value, either single value or partial
+//           representation if a format is passed (yaml / json).
+//   Format: either single, json, or yaml (default: single).
+func (c *Config) Set(key, value, format string) error {
+	if key == "" {
+		return fmt.Errorf("key field must not be empty")
+	}
+	props := strings.Split(key, ".")
+	rv := reflect.ValueOf(c).Elem()
+	found, other, err := getField(props, rv)
+	if err != nil {
+		return err
+	}
+	isOther := other != reflect.Value{}
+
+	field := found
+	if isOther {
+		field = other
+	}
+
+	if field.CanAddr() {
+		i := field.Addr().Interface()
+		in := value
+		switch strings.ToLower(format) {
+		case "yaml", "single", "":
+			if isOther {
+				p := props[len(props)-1]
+				in = fmt.Sprintf("%s: %s", p, value)
+			}
+			err = yaml.Unmarshal([]byte(in), i)
+			if err != nil {
+				return err
+			}
+			return nil
+		case "json":
+			if isOther {
+				p := props[len(props)-1]
+				in = fmt.Sprintf("{%q: %q}", p, value)
```

Comment:
What is `value` here? What happens if I'm trying to set an object? This looks like it will quote my object and then make it a string => string, rather than string => object?

Reply:
I would expect this to be `fmt.Sprintf("{%q: %s}"`, essentially -- confused why it isn't.